Readme.md

```diff
@@ -1,25 +1,111 @@
+
+    ```aspx
+    <dx:ASPxGridView ID="Grid" runat="server" DataSourceID="GridDataSource" ClientInstanceName="Grid">
+        <!-- ... -->
+        <Templates>
+            <GroupRowContent>
+                <dx:ASPxCheckBox ID="checkBox" runat="server" />
+                <dx:ASPxLabel ID="CaptionText" runat="server" Text='<%# GetCaptionText(Container) %>' />
+            </GroupRowContent>
+        </Templates>
+    </dx:ASPxGridView>
+    ```
 
+    ```cs
+    protected string GetCaptionText(GridViewGroupRowTemplateContainer container) {
+        string captionText = !string.IsNullOrEmpty(container.Column.Caption) ? container.Column.Caption : container.Column.FieldName;
+        return string.Format("{0} : {1} {2}", captionText, container.GroupText, container.SummaryText);
+    }
+    ```
+
+3. Handle the Grid View control's [HtmlRowPrepared](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.HtmlRowPrepared?p=netframework) event. In the event handler, access the current group row's check box and specify its [Checked](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxCheckBox.Checked?p=netframework) property. Set the [CheckedChanged](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxCheckBox.CheckedChanged) event of the check box to a function that sends the visible index of the group row and the check state of the check box as callback parameters to the server:
+
+    ```cs
+    protected void Grid_HtmlRowPrepared(object sender, ASPxGridViewTableRowEventArgs e) {
+        if(e.RowType == GridViewRowType.Group) {
+            ASPxCheckBox checkBox = Grid.FindGroupRowTemplateControl(e.VisibleIndex, "checkBox") as ASPxCheckBox;
+            if(checkBox != null) {
+                checkBox.ClientSideEvents.CheckedChanged = string.Format("function(s, e){{ Grid.PerformCallback('{0};' + s.GetChecked()); }}", e.VisibleIndex);
+                checkBox.Checked = GetChecked(e.VisibleIndex);
+            }
+        }
+    }
+
+    protected bool GetChecked(int visibleIndex) {
+        for(int i = 0; i < Grid.GetChildRowCount(visibleIndex); i++) {
+            bool isRowSelected = Grid.Selection.IsRowSelectedByKey(Grid.GetChildDataRow(visibleIndex, i)["ProductID"]); 
+            if(!isRowSelected)
+                return false;
+        }
+        return true;
+    }
+    ```
+
+4. Handle the Grid View's [CustomCallback](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.CustomCallback?p=netframework) event to process the callback. Call the Grid View's [GetChildDataRow](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.GetChildDataRow(System.Int32-System.Int32)?p=netframework) method to get a data row that belongs to the current group. Pass the data row's key and the the check state of the checkbox to the [SetSelectionByKey](https://docs.devexpress.com/AspNet/DevExpress.Web.Data.WebDataSelection.SetSelectionByKey(System.Object-System.Boolean)?p=netframework) method to select or deselect this row:
```

Comment:
> the the check state

Another typo. As you can see, not all typos are highlighted by a spellchecker. Such typos most often come from lengthy editing sessions of a complex sentence or paragraph. I recommend taking another look once your work is complete.

```suggestion
4. Handle the Grid View's [CustomCallback](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.CustomCallback?p=netframework) event to process the callback. Call the Grid View's [GetChildDataRow](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.GetChildDataRow(System.Int32-System.Int32)?p=netframework) method to get a data row that belongs to the current group. Pass the data row's key and the check state of the checkbox to the [SetSelectionByKey](https://docs.devexpress.com/AspNet/DevExpress.Web.Data.WebDataSelection.SetSelectionByKey(System.Object-System.Boolean)?p=netframework) method to select or deselect this row:
```